packages/cosmwasm/src/cw3cosmwasmclient.spec.ts

```diff
@@ -0,0 +1,323 @@
+      const client = new Cw3CosmWasmClient(launchpad.endpoint, alice.address0, wallet, contractAddress);
+      const result = await client.listVoters();
+
+      expect(result.voters.length).toEqual(3);
+      expect(result.voters).toEqual(
+        jasmine.arrayContaining([
+          { addr: alice.address0, weight: 1 },
+          { addr: alice.address1, weight: 1 },
+          { addr: alice.address2, weight: 1 },
+        ]),
+      );
+    });
+  });
+
+  describe("Proposal lifecycle", () => {
+    it("proposal is accepted (proposer has enough weight alone)", async () => {
+      pendingWithoutLaunchpad();
+      pendingWithoutCw3();
+      const contractAddress = deployedCw3.instances[0];
+      const wallet = await Secp256k1HdWallet.fromMnemonic(alice.mnemonic);
+      const client = new Cw3CosmWasmClient(launchpad.endpoint, alice.address0, wallet, contractAddress);
+      const toAddress = makeRandomAddress();
+      const msg = {
+        bank: {
+          send: {
+            from_address: contractAddress,
+            to_address: toAddress,
+            amount: [
+              {
+                amount: "1",
+                denom: "ucosm",
+              },
+            ],
+          },
+        },
+      };
+      await client.createMultisigProposal("My proposal", "A proposal to propose proposing proposals", [msg]);
+      const { proposals } = await client.reverseProposals({ limit: 1 });
+      const proposalId = proposals[0].id;
+      const executeResult = await client.executeMultisigProposal(proposalId);
+      expect(executeResult).toBeTruthy();
+    });
+
+    it("proposal is accepted (proposer does not have enough weight alone)", async () => {
+      pendingWithoutLaunchpad();
+      pendingWithoutCw3();
+      const contractAddress = deployedCw3.instances[1];
+      const proposerWallet = await Secp256k1HdWallet.fromMnemonic(alice.mnemonic);
+      const proposer = new Cw3CosmWasmClient(
+        launchpad.endpoint,
+        alice.address0,
+        proposerWallet,
+        contractAddress,
+      );
+      const voterWallet = await Secp256k1HdWallet.fromMnemonic(alice.mnemonic, makeCosmoshubPath(1));
+      const voter = new Cw3CosmWasmClient(launchpad.endpoint, alice.address1, voterWallet, contractAddress);
+      const toAddress = makeRandomAddress();
+      const msg = {
+        bank: {
+          send: {
+            from_address: contractAddress,
+            to_address: toAddress,
+            amount: [
+              {
+                amount: "1",
+                denom: "ucosm",
+              },
+            ],
+          },
+        },
+      };
+      await proposer.createMultisigProposal("My proposal", "A proposal to propose proposing proposals", [
+        msg,
+      ]);
+      const { proposals } = await voter.reverseProposals({ limit: 1 });
+      const proposalId = proposals[0].id;
+      const voteResult = await voter.voteMultisigProposal(proposalId, Vote.Yes);
```

Comment:
I would test that trying to execute before this second vote is falsy. To ensure we are covering this case.